blocks/user-licenses/user-licenses.js

```diff
@@ -1,5 +1,400 @@
+    const data = await execute('dynamics_set_license_nickname', `&id=97d53652-122f-e611-80ea-005056831cd4&nickname=${value}`, 'PATCH');
+    console.log('Updated license nickname ', data);
+  } else {
+    // TODO handle response and add toast message
+    alert('Provide nickname value before saving');
+  }
+}
+
+function hideOtherUserLicenseInformation() {
+  const sections = document.querySelectorAll('.section');
+
+  sections.forEach((section) => {
+    if (section.classList.contains('user-licenses-container')) {
+      const tabsContent = section.querySelector('.tabs');
+      if (tabsContent !== undefined && tabsContent !== null) {
+        section.querySelector('.tabs').remove();
+      }
+      const licenseComponent = section.querySelector('.user-licenses.block');
+      licenseComponent.querySelector('h3').parentElement.parentElement.remove();
+
+      let backToProfileButton = licenseComponent.querySelector('#backToAccountButton');
+      if (backToProfileButton === undefined || backToProfileButton === null) {
+        backToProfileButton = createTag('a', { href: '/pages/profile', class: 'button primary', id: 'backToAccountButton' }, 'Back to Account');
+        licenseComponent.insertBefore(backToProfileButton, licenseComponent.firstChild);
+      }
+    } else {
+      section.remove();
+    }
+  });
+}
+
+async function displayLicenseDetails(event) {
+  hideOtherUserLicenseInformation();
+  const licenseId = event.target.getAttribute('data-license-id');
+  const userId = localStorage.getItem('auth0_id');
+  const accessToken = localStorage.getItem('accessToken');
+
+  if (userId == null || userId === undefined || accessToken == null || accessToken === undefined) {
+    window.location.assign(`${window.location.origin}`);
+  } else {
+    const data = await execute('dynamics_get_end_users_for_license', `&license_id=${licenseId}`, 'GET');
+    // DOM creation
+    const manageLicenseH2 = createTag('h2', '', `Manage License #${data.licenseid}`);
+    const licenseDetailsDiv = document.querySelector('.license-details');
+    licenseDetailsDiv.innerHTML = '';
+    licenseDetailsDiv.appendChild(manageLicenseH2);
+    const licenseDetailsDataDiv = createTag('div', { class: 'license-details-data' }, '');
+    licenseDetailsDiv.appendChild(licenseDetailsDataDiv);
+    const headings = ['License Administrator|_new_registereduser_value@OData.Community.Display.V1.FormattedValue', 'Account|_new_account_value@OData.Community.Display.V1.FormattedValue', 'Renewal Date|new_supportexpires@OData.Community.Display.V1.FormattedValue', 'Key Serial Number|new_licenseid', 'Product|_new_product_value@OData.Community.Display.V1.FormattedValue', 'License Type|zemax_seattype@OData.Community.Display.V1.FormattedValue', 'ZPA Support|new_premiumsupport@OData.Community.Display.V1.FormattedValue', 'Seat Count|new_usercount@OData.Community.Display.V1.FormattedValue', 'End User Count|new_endusercount@OData.Community.Display.V1.FormattedValue'];
```

Comment:
I am not sure what's happening here. what is the meaning of the `|`? Maybe it should be encoded as a JSON object?